File: website_portal_debt_notebook/static/src/js/test_website_portal_debt_notebook.js
/*  Copyright 2019 Anvar Kildebekov <https://it-projects.info/team/fedoranvar>
    License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html). */
odoo.define('website_portal_debt_notebook.tour', function (require) {
    "use strict";

    var tour = require("web_tour.tour");
    var core = require('web.core');
    var base = require("web_editor.base");
    var _t = core._t;

    function check_debt_history(journal, cash) {
        return [{
            trigger: "a[href*='/my/debt_history']:first",
            content: _t("<p>Click to enter 'debt history'</p>"),
            position: "bottom",
        },{
            trigger: ".history-line:nth-child(2)",
            content: _t("<p>Check credit-lines</p>"),
            position: "bottom",
            run: function(){
                for (var i = 0; i < 2; i++) {
                    var line_journal = $(".history-line:nth-child(" + (i+1).toString() + ")>td:nth-child(2)")[0].innerText;
                    var line_cash = $(".history-line:nth-child(" + (i+1).toString() + ")>td:nth-child(4)")[0].innerText;
                    if (line_journal !== journal || line_cash !== cash[i]) {
                        console.log(i + '-line failed: ' + line_journal + ' || ' + line_cash + '\n');
                        console.log('error');
                        break;
                    }
                }
            }
        }];
    }

    var steps = [];
    var journal = "Test (USD)";
    var cash = ["10.0",
                "1.0"];
    steps = steps.concat(check_debt_history(journal, cash));

    tour.register('website_portal_debt_notebook_tour', { test: true, url: '/my', wait_for: base.ready()}, steps);
});

File: website_portal_debt_notebook/tests/test_website_portal_debt_notebook.py
# Copyright 2019 Anvar Kildebekov <https://it-projects.info/team/fedoranvar>
# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).

import odoo.tests


@odoo.tests.common.at_install(True)
@odoo.tests.common.post_install(True)
class TestUi(odoo.tests.HttpCase):

    def common_test(self, test_login):

        env = self.env
        #  For QWeb-loading
        env['ir.module.module'].search([('name', '=', 'website_portal_debt_notebook')], limit=1).state = 'installed'
        #  Step 1:
        #  Set test-variables: partner_id, journal_id, currency_id
        balance = ["1.0",
                   "10.0"]
        now_date = ["01/01/2019 11:11:11",
                    "01/01/2019 22:22:22"]
        test_partner = env['res.users'].search([('login', '=', test_login)]).partner_id
        test_currency_id = env['res.currency'].search([('name', '=', 'USD')]).id
        #  Create journal
        test_journal = env['account.journal'].create({
            'name': "Test",
            'type': 'cash',
            'journal_user': True,
            'debt': True,
            'code': 'PLKS',
        })
        #  Create credit-records
        for i in range(2):
            env['pos.credit.update'].create({
                'partner_id': test_partner.id,
                'date': now_date[i],
                'journal_id': test_journal.id,
                'currency_id': test_currency_id,
                'balance': balance[i],
            }).switch_to_confirm()
        #  Step 2:
        #  run tour
        self.browser_js("/",
                        "odoo.__DEBUG__.services['web_tour.tour'].run('website_portal_debt_notebook_tour', 500)",
                        "odoo.__DEBUG__.services['web_tour.tour'].tours.website_portal_debt_notebook_tour.ready",
                        login=test_login)

    def test_01_website_portal_debt_notebook(self):
        self.common_test("portal")
